File: client/bindings/python/examples/stronghold.py
from iota_client import IotaClient, StrongholdSecretManager

# Create an IotaClient instance
client = IotaClient({'nodes': ['https://api.testnet.shimmer.network']})

secret_manager = StrongholdSecretManager("client.stronghold", "some_hopefully_secure_password")

# Store the mnemonic in the Stronghold snapshot, this needs to be done only the first time.
# The mnemonic can't be retrieved from the Stronghold file, so make a backup in a secure place!
result = client.store_mnemonic(secret_manager, "flame fever pig forward exact dash body idea link scrub tennis minute " +
    "surge unaware prosper over waste kitten ceiling human knife arch situate civil")

# Generate public address with custom account index and range.
address = client.generate_addresses(secret_manager, {
    "account_index": 0,
    "range": {
        "start": 0,
        "end": 1,
    },
})

print(f'Address: {address[0]}')

File: client/bindings/nodejs/examples/stronghold.ts
// Copyright 2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { Client, initLogger } from '@iota/client';
require('dotenv').config({ path: '../.env' });

// Run with command:
// node ./dist/stronghold.js

// In this example we will store a seed in the Stronghold vault and generate an address
async function run() {
    initLogger();
    if (!process.env.NODE_URL) {
        throw new Error('.env NODE_URL is undefined, see .env.example');
    }

    const client = new Client({
        // Insert your node URL in the .env.
        nodes: [process.env.NODE_URL],
    });

    try {
        if (!process.env.STRONGHOLD_PASSWORD) {
            throw new Error(
                '.env stronghold password is undefined, see .env.example',
            );
        }
        if (!process.env.NON_SECURE_USE_OF_DEVELOPMENT_MNEMONIC_1) {
            throw new Error('.env mnemonic is undefined, see .env.example');
        }
        const secretManager = {
            stronghold: {
                password: process.env.STRONGHOLD_PASSWORD,
                snapshotPath: 'client.stronghold',
            },
        };

        // Store the mnemonic in the Stronghold snapshot, this needs to be done only the first time.
        // The mnemonic can't be retrieved from the Stronghold file, so make a backup in a secure place!
        await client.storeMnemonic(
            secretManager,
            process.env.NON_SECURE_USE_OF_DEVELOPMENT_MNEMONIC_1,
        );

        const address = await client.generateAddresses(secretManager, {
            accountIndex: 0,
            range: {
                start: 0,
                end: 1,
            },
        });

        console.log('First public address:', address, '\n');
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
